### ISSUE TITLE
[Feature Request] Support dmypy `--status-file` flag

### ISSUE THREAD

**Issue by Jasha10** [python-lsp/pylsp-mypy#47]
Hello,

I'd like to request support for setting the [dmypy `--status-file`](https://mypy.readthedocs.io/en/stable/mypy_daemon.html#cmdoption-dmypy-status-file) flag via the `pylsp-mypy.cfg` file.

The use case is to ensure that multiple instances of the dmypy server do not collide with eachother.

##### Desired behavior:
Setting e.g. `"dmypy_status_file": ".dmypy-pylsp.json"` in the `pylsp-mypy.cfg` file should cause this plugin's invocations of `dmypy` to pass an argument `--status-file=.dmypy-pylsp.json`.

**Comment by Jasha10** [comment]
One might think that the `overrides` key in the `.pylsp-mypy.cfg` could handle this use-case, but it does not, because the overrides are not consistently applied for all calls to dmypy.
For example, the calls [`mypy_api.run_dmypy(["status"])`](https://github.com/python-lsp/pylsp-mypy/blob/81cf8f4897871e5f57dd653c4ae241a34c135cb0/pylsp_mypy/plugin.py#L256) and [`mypy_api.run_dmypy(["kill"])`](https://github.com/python-lsp/pylsp-mypy/blob/81cf8f4897871e5f57dd653c4ae241a34c135cb0/pylsp_mypy/plugin.py#L263) in this plugin's `pylsp_lint` function do not take into account the overrides.

The `--status-file` argument would need to be passed in consistently across all calls to dmypy.

**Comment by Richardk2n** [comment]
So passing the overrides consistently would also resolve this?

**Comment by Jasha10** [comment]
I do not think the overrides should be passed consistently.

The overrides work with `dmypy run -- src` but they will not always work with `dmypy status` and `dmypy kill`.

For example, `dmypy run -- src --check-untyped-defs` makes sense but `dmypy kill --check-untyped-defs` does not make sense.

We would need to implement a separate `dmypy_overrides` configuration parameter that gets passed to `dmypy` specifically.

**Comment by Jasha10** [comment]
@Richardk2n I'd be happy to open a PR implementing `dmypy_overrides`.

**Comment by Richardk2n** [comment]
Are there such overrides other than the status file that are relevant and are passed in this matter?

(asking for if we should implement and overrides dict or just the flag)

**Comment by Jasha10** [comment]
Good question. After looking through the other dmypy flags ([here's a full list](https://mypy.readthedocs.io/en/stable/mypy_daemon.html#additional-daemon-flags)), I think only `--status-file` makes sense to pass in this manner.

- The `--log-file` flag _could_ be passed in this manner, but `--log-file` controls the destination of stdout/stderr. Passing `--log-file` via pylsp-mypy doesn't make sense since pylsp-mypy already handles dmypy's output streams itself.
- All the other flags (besides `--status-file` and `--log-file`) appear to be supported by only a _subset_ of the dmypy commands, e.g. the `--timeout` flag only works with `dmypy start` / `dmypy restart` / `dmypy run`. This means e.g. applying `--timeout=10` to `dmypy kill` or `dmypy status` wouldn't make sense. Thus `--timeout` would not work with an overrides dict.

In conclusion I think that, given `dmypy`'s currently-supported set of flags, implementing _only_ the `--status-file` flag is the best design.
